zstd_vendor/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+cmake_minimum_required(VERSION 3.5)
+project(zstd_vendor)
+
+list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
+find_package(zstd)
+
+if (NOT zstd_FOUND)
+  set(extra_cmake_args)
+
+  if(DEFINED CMAKE_TOOLCHAIN_FILE)
+    list(APPEND extra_cmake_args "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
+  endif()
+
+  include(ExternalProject)
+  set(EP_PREFIX zstd-1.4.4)
+  ExternalProject_Add(zstd-1.4.4
+    GIT_REPOSITORY https://github.com/facebook/zstd.git
```

Comment:
i'm thinking it'd be better to use a release tarball rather than a git tag, the other vendors do it, so for consistency at least, e.g. https://github.com/facebook/zstd/releases/download/v1.4.4/zstd-1.4.4.tar.gz - and do an MD5